bika/lims/browser/templates/analysisservice_popup.pt

```diff
@@ -87,7 +87,7 @@
              tal:condition="filename"
              tal:define="file method/MethodDocument;
                          filename python:getattr(file, 'filename', '');
-                         filesize python:file.getObjSize();
+                         filesize python:file.get_size() or 0;
```

Comment:
I've checked `getObjSize()` from `Products.CMFPlone.CatalogTool`and the function already delegates the action to `get_size`:

```python
def getObjSize(obj):
    """ Helper method for catalog based folder contents.
    """
    smaller = SIZE_ORDER[-1]

    if base_hasattr(obj, 'get_size'):
        size = obj.get_size()
    else:
        size = 0
    ...
```

Also, as per the following log that appears in the traceback, I am not sure this change would be enough:
```
Unauthorized: The object is marked as private.  Access to 'get_size' of (plone.app.blob.field.BlobWrapper object at 0x10ce0f938) denied.
```

I am missing something?

Reply:
`plone.app.blob.field.BlobWrapper` defines this:
```
    @security.private
    def get_size(self):
        """ return the size of the blob """
        blob = openBlob(self.blob)
        size = fstat(blob.fileno()).st_size
        blob.close()
        return size
```

Afaik this means that it is not accessible from a template or script context.

So if you analyze this part of the traceback, it makes sense:

```
 Module script, line 17, in getObjSize
   - <FSPythonScript at /bikalims/methods/method-5/getObjSize>
   - Line 17
```